Update contributing docs for sass modules (#7055)
alphagov/govuk-frontend: docs/contributing/coding-standards/components.md

When creating your component, you should create the following files in the component’s folder:

- `README.md` - Summary documentation with links to the installation instructions and component documentation on <https://design-system.service.gov.uk/>

> - `_index.scss` - The main Sass entry point for the component. Generates the styles by including the mixin from the `_mixin.scss` file. You can include it [on its own](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-specific-parts-of-the-css), or [alongside other components in `components/_index.scss`](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-specific-parts-of-the-css)

Reviewer: ```suggestion
- `_index.scss` - The main Sass entry point for the component. Generates the styles by including the mixin from the `_mixin.scss` file. You can include `_index.scss` [on its own](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-specific-parts-of-the-css), or [alongside other components in `components/_index.scss`](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-specific-parts-of-the-css)
```

> - `_mixin.scss` - The actual styles for the component, contained within a mixin. The mixin is included by the `_index.scss` file. This file should not be included directly.

Reviewer: ```suggestion
- `_mixin.scss` - The actual styles for the component, contained within a mixin. The `_index.scss` file includes the mixin. You should not include the `_mixin.scss`  file directly.
```

- `[component-name].unit.test.mjs` - Unit tests to verify any component-specific lower-level logic.
- `[component-name].test.js` - Functional tests to verify the behaviour of the whole component

> In version 7 of GOV.UK Frontend, we will remove support for `@import` in Sass. Before then, you must include the following files to continue supporting `@import`:

Reviewer: ```suggestion
In version 7 of GOV.UK Frontend, we'll remove support for `@import` in Sass. Before then, you must include the following files to continue supporting `@import`:
```
